check_command.go

```diff
@@ -75,6 +77,27 @@ func (c *CheckCommand) Run(request CheckRequest) ([]Version, error) {
 			continue
 		}
 
+		if request.Source.SemverConstraint != "" {
+			constraint, err := semver.NewConstraint(request.Source.SemverConstraint)
+			if err != nil {
+				return []Version{}, err
+			}
```

Comment:
nit: can the constraint be constructed outside of the for loop? While it's very low cost, `semver.NewConstraint` does runs a `regexp.FindAllStringSubmatch` every single time

Reply:
for sure, fixed